File: Niveau 3/Niveau 3.11/carres.py
#-------------------------------------------------------------------------------
# Name:        Carrés concentriques
# Purpose:     France-IOI niveau 3.11
#
# Author:      Sébastien
#
# Created:     23/10/2022
#-------------------------------------------------------------------------------

def distance(n, x, y):
    return max(abs(n-x), abs(n-y))

def main():
    n = int(input())
    mot = input()

    for y in range(1, 2*n):
        ligne = [mot[distance(n, x, y)] for x in range(1, 2*n)]
        print("".join(ligne))

if __name__ == '__main__':
    main()

"""
5
carre
"""

File: Niveau 3/Niveau 3.11/carres.hs
--------------------------------------------------------------------------------
-- Name:        Carrés concentriques
-- Purpose:     France-IOI niveau 3.11
--
-- Author:      Sébastien
--
-- Created:     23/10/2022
--------------------------------------------------------------------------------

import Control.Monad
import Data.Array

main :: IO ()
main = do
    n <- readLn :: IO Int
    mot <- fmap (listArray (0, n-1)) getLine
    let carre = foldr (\y ys -> foldr (afficherCaractere mot y) "" [(1-n)..(n-1)]:ys) [] [(1-n)..(n-1)]
    putStr $ unlines carre
    
afficherCaractere :: Array Int Char -> Int -> Int -> String -> String
afficherCaractere mot y x xs = (mot!(max (abs x) (abs y))):xs